core-rust/state-manager/src/metrics.rs

```diff
@@ -111,6 +130,89 @@ impl LedgerMetrics {
+            )
+            .registered_at(registry),
+            execution_cost_units_consumed: new_histogram(
+                opts(
+                    "committed_transactions_execution_cost_units_consumed",
+                    "Execution cost units consumed per committed transactions.",
+                ),
+                Self::buckets(DEFAULT_COST_UNIT_LIMIT as usize),
+            )
+            .registered_at(registry),
+            substate_read_size: new_histogram(
+                opts(
+                    "committed_transactions_substate_read_size",
+                    "Total (per committed transaction) substate read size in bytes.",
+                ),
+                // TODO: update once max substate reads can be limited at execution
+                Self::buckets(DEFAULT_MAX_TOTAL_VERTEX_SUBSTATE_READ_SIZE),
+            )
+            .registered_at(registry),
+            substate_read_count: new_histogram(
+                opts(
+                    "committed_transactions_substate_read_count",
+                    "Number of substate reads per committed transactions.",
+                ),
+                Self::buckets(DEFAULT_MAX_SUBSTATE_READS_PER_TRANSACTION),
+            )
+            .registered_at(registry),
+            substate_write_size: new_histogram(
+                opts(
+                    "committed_transactions_substate_write_size",
+                    "Total (per committed transaction) substate write size in bytes.",
+                ),
+                // TODO: update once max substate writes can be limited at execution
+                Self::buckets(DEFAULT_MAX_TOTAL_VERTEX_SUBSTATE_WRITE_SIZE),
+            )
+            .registered_at(registry),
+            substate_write_count: new_histogram(
+                opts(
+                    "committed_transactions_substate_write_count",
+                    "Number of substate writes per committed transactions.",
+                ),
+                Self::buckets(DEFAULT_MAX_SUBSTATE_WRITES_PER_TRANSACTION),
+            )
+            .registered_at(registry),
+            max_wasm_memory_used: new_histogram(
+                opts(
+                    "committed_transactions_max_wasm_memory_used",
+                    "Maximum WASM memory used in bytes per committed transaction.",
+                ),
+                Self::buckets(DEFAULT_MAX_WASM_MEM_PER_TRANSACTION),
+            )
+            .registered_at(registry),
+            max_invoke_payload_size: new_histogram(
+                opts(
+                    "committed_transactions_max_invoke_payload_size",
+                    "Maximum invoke payload size in bytes per committed transaction.",
+                ),
+                Self::buckets(DEFAULT_MAX_INVOKE_INPUT_SIZE),
+            )
+            .registered_at(registry),
+        }
+    }
+
+    // Given a limit, builds buckets for a Histogram with higher resolution for lower values.
+    fn buckets(limit: usize) -> Vec<f64> {
```

Comment:
_(question from author)_

Let me know if this seems reasonable enough (or if hand-picked buckets are preferred).

Reply:
Although might be worth a quick comment?